src/Resources/contao/dca/tl_user.php

```diff
@@ -634,9 +634,14 @@ public function deleteUser($row, $href, $label, $title, $icon, $attributes)
 	 * @return string
 	 *
 	 * @throws Exception
+	 *
+	 * @deprecated Deprecated since Contao 4.x, to be removed in Contao 5.0.
+	 *             Use the switch_user_button_generator service instead.
 	 */
 	public function switchUser($row, $href, $label, $title, $icon)
 	{
+		@trigger_error('Using tl_user->switchUser() has been deprecated and will no longer work in Contao 5.0. Use the switch_user_button_generator service instead', E_USER_DEPRECATED);
+
```

Comment:
The rest of this method should be removed or replaces by the new switch user stuff.